scripts/monte_carlo.py

```diff
@@ -177,71 +370,79 @@ def monte_carlo_sampling_scipy(
     configure_logging(snakemake)
     monte_carlo_config = snakemake.params.monte_carlo
 
-    ### SCENARIO INPUTS
+    # SCENARIO INPUTS
     ###
-    MONTE_CARLO_PYPSA_FEATURES = {
-        k: v for k, v in monte_carlo_config["pypsa_standard"].items() if v
-    }  # removes key value pairs with empty value e.g. []
+    MONTE_CARLO_PYPSA_FEATURES = [
+        k for k in monte_carlo_config["uncertainties"].keys() if k
+    ]  # removes key value pairs with empty value e.g. []
     MONTE_CARLO_OPTIONS = monte_carlo_config["options"]
-    L_BOUNDS = [item[0] for item in MONTE_CARLO_PYPSA_FEATURES.values()]
-    U_BOUNDS = [item[1] for item in MONTE_CARLO_PYPSA_FEATURES.values()]
     N_FEATURES = len(
         MONTE_CARLO_PYPSA_FEATURES
     )  # only counts features when specified in config
     SAMPLES = MONTE_CARLO_OPTIONS.get(
         "samples"
     )  # TODO: What is the optimal sampling? Fabian Neumann answered that in "Broad ranges" paper
     SAMPLING_STRATEGY = MONTE_CARLO_OPTIONS.get("sampling_strategy")
+    UNCERTAINTIES_VALUES = monte_carlo_config["uncertainties"].values()
+    SEED = MONTE_CARLO_OPTIONS.get("seed")
+
+    # PARAMETERS VALIDATION
+    # validates the parameters supplied from config file
+    validate_parameters(SAMPLING_STRATEGY, SAMPLES, UNCERTAINTIES_VALUES)
 
-    ### SCENARIO CREATION / SAMPLING STRATEGY
+    # SCENARIO CREATION / SAMPLING STRATEGY
     ###
     if SAMPLING_STRATEGY == "pydoe2":
         lh = monte_carlo_sampling_pydoe2(
             N_FEATURES,
             SAMPLES,
+            UNCERTAINTIES_VALUES,
+            random_state=SEED,
             criterion=None,
             iteration=None,
-            random_state=42,
             correlation_matrix=None,
         )
     if SAMPLING_STRATEGY == "scipy":
         lh = monte_carlo_sampling_scipy(
             N_FEATURES,
             SAMPLES,
+            UNCERTAINTIES_VALUES,
+            seed=SEED,
             centered=False,
             strength=2,
             optimization=None,
-            seed=42,
         )
     if SAMPLING_STRATEGY == "chaospy":
         lh = monte_carlo_sampling_chaospy(
-            N_FEATURES,
-            SAMPLES,
-            rule="latin_hypercube",
-            seed=42,
+            N_FEATURES, SAMPLES, UNCERTAINTIES_VALUES, seed=SEED, rule="latin_hypercube"
         )
-    lh_scaled = qmc.scale(lh, L_BOUNDS, U_BOUNDS)
 
-    ### MONTE-CARLO MODIFICATIONS
+    # create plot for the rescaled distributions
+    for idx in range(N_FEATURES):
+        sns.displot(lh[:, idx], kde=True).set(
+            title=f"{MONTE_CARLO_PYPSA_FEATURES[idx]}"
+        ).figure.savefig(f"{MONTE_CARLO_PYPSA_FEATURES[idx]}.png", bbox_inches="tight")
```

Comment:
This is a debugging code that I don't dislike to keep as it may be useful for development purposes, but I'd keep it commented.
Accordingly, I'd revice the title to highlight that, e.g. "# create plot for the rescaled distributions (for development usage, commented by default)"

Reply:
I have commented this part out in the recent commits.